dali/python/nvidia/dali/pipeline.py

```diff
@@ -769,6 +769,15 @@ def feed_input(self, data_node, data, layout = None, cuda_stream = None, use_cop
             _data_node._check(data_node)
             name = data_node.name
 
+        # Check if user uses feed_input on an external_source operator that was initialized with 'source'
+        if next((op._callback is not None for op in self._ops if op.name == name), False):
+            _, _, _, values = inspect.getargvalues(inspect.currentframe().f_back)
+            instance = values.get('self', None)
+            class_obj = getattr(instance, '__class__', None) if instance else None
+            if not class_obj or getattr(class_obj, '__name__', None) != '_ExternalSourceGroup':
+                raise RuntimeError(f"Cannot feed input data to the external source '{name}'. "
+                                   "It was already populated with data through 'source' parameter.")
```

Comment:
It's OK, however, I'd simplify the message to something like:
```suggestion
                raise RuntimeError(f"Cannot use `feed_input` on the external source '{name}' with a `source` argument specified.")

```

Reply:
sure, done